docs/API_DESIGN.md

```diff
@@ -90,11 +92,29 @@ CLASS VerifiableCredential
   PUBLIC DATA credentialSubject: CredentialSubject
 
   CONSTRUCTOR create(issuer: Issuer, credential_subject: CredentialSubject, options: CreateOptions)
+  CONSTRUCTOR decode(vc_jwt: string, options: VerifiableCredentialDecodeOptions)
 
-  CONSTRUCTOR(vcjwt: string)
-  CONSTRUCTOR(vcjwt: string, verifier: Verifier)
-  METHOD sign(bearer_did: BearerDid): string
-  METHOD sign_with_signer(key_id: string, signer: Signer): string
+  METHOD sign(options: VerifiableCredentialSignOptions)
+```
+
+##### `VerifiableCredentialDecodeOptions`
+
+```pseudocode!
+CLASS VerifiableCredentialDecodeOptions
+  PUBLIC DATA verify: bool?
```

Comment:
I usually prefer a modular approach of separate decode() and verify() methods. If we have to combine it, then it seems clearer to use a `verify: bool` instead of `bool?`. If we have to use `bool?`, what is the behavior of `verify == null`? Does it verify?

If we have only one option, why not have it as a param, rather than defining a struct that wraps a boolean?

Reply:
Is it only possible to verify open calling `decode()`? Or do we plan to also add a separate `verify()` method later, to allow someone to verify after decoding?